Refactor/separate methods in classes

jExbrayat opened the pull request:
Split the single class detrend in detrend.py into several classes.

Create _fancy_plot function in detrend_fancy_plot.py.
It is meant to be called in the fancy_plot method in each detrend class.

jExbrayat committed:
refactor: rename detrend.py to old_detrend.py

jExbrayat committed:
feat: create LinearReg class

jExbrayat committed:
refactor: create external _fancy_plot function

jExbrayat committed:
refactor: put self variable in _fancy_plot function's argument instead of string

jExbrayat committed:
feat: create PolynomialRegression class

jExbrayat committed:
feat: create LinearMA class

jExbrayat committed:
feat: create ExponentialMA class

jExbrayat committed:
feat: create BSplines class

jExbrayat committed:
refactor: delete useless imports

jExbrayat committed:
chore: delete old_detrend.py

jExbrayat committed:
refactor: adapt and clean demo notebook

Ab2nour commented on src/detrend_fancy_plot.py:
```diff
@@ -0,0 +1,48 @@
+    xticklabels: pd.core.indexes.base.Index | None = None,
+    method_name: str = "",
+) -> None:
+    """plot two graphs : the original data and its fitted trend curve ; the detrended data
```
nit: "Plot two graphs: the original [...]"

=> uppercase letter at the start of docstring
=> no space before ":"
=> also, no space before ";"

(nit means nitpicking => le mec qui clc quoi, et du coup tu préfixes de "nit" pour dire que tu attaques sur des détails, pour ne pas contrarier l'autre. Car il faut quand même pointer tous les petits détails, ça reste important)

jExbrayat replied:
done boi

Ab2nour replied:
you forgot to commit boi

Ab2nour commented on src/detrend.py:
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
+    def fit(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
         """
-        returns fitted values with the linear centered mobile average method
+        returns fitted values with the linear mobile average method
```
nit: uppercase at the start of docstring "Returns fitted..."

jExbrayat replied:
done

Ab2nour commented on src/detrend.py:
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
+        return self.y_predict
+
+    def fancy_plot(self, xticklabels: pd.core.indexes.base.Index | None = None) -> None:
+        """plot two graphs : the original data and its fitted trend curve ; the detrended data
```
nit: uppercase at start of docstring + no space before ":" and ";"

Ab2nour commented on src/detrend.py:
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
+        return self.y_predict
+
+    def fancy_plot(self, xticklabels: pd.core.indexes.base.Index | None = None) -> None:
+        """plot two graphs : the original data and its fitted trend curve ; the detrended data
```
nit: same

jExbrayat replied:
done

Ab2nour commented on src/detrend.py:
```diff
@@ -102,39 +112,135 @@ def _PolynomialRegression(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
 
+    def fit(self, y: np.ndarray | pd.DataFrame) -> np.ndarray:
+        """
+        returns fitted values with the B-splines method
```
nit: "Returns"

jExbrayat replied:
done

Ab2nour reviewed (changes requested):
Only nitpicking. Otherwise, it's all good

jExbrayat committed:
docs: correct syntax

jExbrayat committed:
docs: correct syntax

jExbrayat committed:
feat: create detrend and write process script

jExbrayat committed:
refactor: delete .csv in stock name and add .csv at end of processed filename

jExbrayat committed:
refactor: execute script for polynomial regression model

jExbrayat committed:
feat: create demo notebook for detrended stock data

jExbrayat committed:
feat: plot candle chart of detrend data against original data

jExbrayat committed:
refactor: improve detrend classes and adapt demo detrend notebook

Add possibility to specify different time series for fit and predict as long as they are the same length.

Useful to detrend stock prices with several values at each time step (open, close, high, low).

jExbrayat committed:
refactor: fit only 1 time series per stock in write_detrend_data and correct filename

The stock data are now detrended using the close price (arbitrary). This method allows coherent candle stick data, takes less time to process and gives an equally correct result.

Processed filenames no longer contain an undesired dot after their stock name.

jExbrayat committed:
refactor: re execute write_detrend_data.py and demo notebook

jExbrayat committed:
feat/writing-detrended-data-to-csv

Create script to write results of detrended data in csv files.

jExbrayat committed:
refactor: change save directory for detrended files

jExbrayat committed:
fix: set right directory to import detrend module

jExbrayat committed:
refactor: reduce code length thanks to native option

jExbrayat committed:
feat: run write_detrend_data.py for LinearMA

jExbrayat committed:
docs: explain model_options format for directory creation

jExbrayat committed:
refactor/change-organization-of-processed-data

Refactor/change organization of processed data

Ab2nour commented:
LGTM!
